packages/gafl-webapp-service/src/handlers/oidc-handler.js

```diff
@@ -0,0 +1,108 @@
+import { salesApi } from '@defra-fish/connectors-lib'
+import { OIDC_SIGNIN } from '../uri.js'
+import Boom from '@hapi/boom'
+import OpenIdClient from 'openid-client'
+import moment from 'moment'
+import debug from 'debug'
+
+const { generators, Issuer } = OpenIdClient
+const db = debug('webapp:oidc')
+
+let client = null
+let cache = null
+let redirectUri = null
+/**
+ * Initialise OpenID connect authentication for the hapi server
+ *
+ * @param server the hapi server instance to configure
+ * @returns {Promise<void>}
+ */
+export const initialise = async server => {
+  const MsIssuer = await Issuer.discover(process.env.OIDC_ENDPOINT)
+  db('Discovered issuer %s %O', MsIssuer.issuer, MsIssuer.metadata)
+
+  redirectUri = new URL(OIDC_SIGNIN.uri, process.env.OIDC_REDIRECT_HOST).toString()
+  client = new MsIssuer.Client({
+    client_id: process.env.OIDC_CLIENT_ID,
+    client_secret: process.env.OIDC_CLIENT_SECRET,
+    redirect_uris: [redirectUri],
+    response_types: ['code', 'id_token']
+  })
+  cache = server.cache({ segment: 'oidc', expiresIn: 12 * 60 * 60 * 1000 })
+
+  server.auth.strategy('oidc', 'cookie', {
+    cookie: {
+      name: process.env.OIDC_SESSION_COOKIE_NAME,
+      password: process.env.OIDC_SESSION_COOKIE_PASSWORD,
+      ttl: null,
+      isSecure: process.env.NODE_ENV !== 'development',
+      isHttpOnly: process.env.NODE_ENV !== 'development',
+      isSameSite: 'Lax',
+      path: '/'
+    },
+    redirectTo: oidcRedirect,
+    validateFunc: async (request, session) => {
+      if (session.oid) {
+        return { valid: true, credentials: session }
+      }
+      return { valid: false }
+    }
+  })
+  server.auth.default('oidc')
+}
+
+const oidcRedirect = request => {
+  const nonce = generators.nonce()
+  const state = generators.state()
+  cache.set(state, { state, nonce, postAuthRedirect: request.path })
+  request.cookieAuth.clear()
+  const url = client.authorizationUrl({
+    scope: 'openid profile email',
+    response_type: 'code id_token',
+    response_mode: 'form_post',
+    domain_hint: 'defra.gov.uk',
+    nonce: nonce,
+    state: state
+  })
+  db('Redirect URL: %s', url)
+  return url
+}
+
+/**
+ * OIDC Authentication Handler - handles POST callbacks from the OpenID connect provider
+ *
+ * @param request the hapi request object
+ * @param h the hapi response handler
+ * @returns {Promise}
+ */
+export const signIn = async (request, h) => {
+  const success = !!request.payload.id_token
+  if (success) {
+    // Retrieve the nonce from the server cache for the given state value
+    const { nonce, state, postAuthRedirect = '/buy' } = (await cache.get(request.payload.state)) ?? {}
```

Comment:
There is a constant for this route in uri.js  for this - CONTROLLER

Reply:
Refactored to use CONTROLLER.uri